src/main/java/gift/application/ProductService.java

```diff
@@ -0,0 +1,73 @@
+package gift.application;
+
+import gift.dao.ProductRepository;
+import gift.domain.Product;
+
+import gift.dto.ProductRequest;
+import gift.dto.ProductResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+
+    private final ProductRepository productRepository;
+
+    @Autowired
+    public ProductService(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    public List<ProductResponse> getAllProducts() {
+        List<Product> productList = productRepository.findAll();
+        List<ProductResponse> responseList = new ArrayList<>();
+        for (Product product : productList) {
+            responseList.add(new ProductResponse(product));
+        }
+        return responseList;
+    }
+
+    public ProductResponse getProductById(Long id) {
+        Product product = productRepository.findById(id).orElseThrow(
+                () -> new IllegalArgumentException("해당 상품은 존재하지 않습니다")
```

Comment:
데이터가 없을 때 던질 예외로 NoSuchElementException 을 사용해보는 것은 어떨까요?!😀
```suggestion
                () -> new NoSuchElementException("해당 상품은 존재하지 않습니다")
```

- NoSuchElementException: 이 예외는 주로 컬렉션이나 데이터베이스에서 특정 요소를 찾을 수 없을 때 사용됩니다.
- IllegalArgumentException: 이 예외는 메서드에 전달된 인수가 적절하지 않을 때 사용됩니다.